<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>524</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Planet Generator</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::ForceTabbedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_10">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QWidget" name="canvas3D" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="shapesDock">
   <property name="windowTitle">
    <string>Planet Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="shapesDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <widget class="QGroupBox" name="shapeParameters">
       <property name="title">
        <string>Planet Parameters</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <property name="verticalSpacing">
         <number>5</number>
        </property>
        <item row="7" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="baseRoughnessLabel">
            <property name="text">
             <string>Base Roughness</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="baseRoughnessSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="baseRoughnessTextbox">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="6" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="roughnessLabel">
            <property name="text">
             <string>Roughness</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="roughnessSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="roughnessTextbox">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="10" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="minValueLabel">
            <property name="text">
             <string>Min Value</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="minValueSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="minValueTextbox">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="noiseLabel">
          <property name="text">
           <string>Noise</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="strengthLabel">
            <property name="text">
             <string>Strength</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="strengthSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="strengthTextbox">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="11" column="0">
         <widget class="QLabel" name="centerLabel">
          <property name="text">
           <string>Center</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="resolutionLabel">
            <property name="text">
             <string>Resolution</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="resolutionSlider">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="resolutionTextbox">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="8" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="numLayersLabel">
            <property name="text">
             <string>Num Layers</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="numLayersSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="numLayersTextbox">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="12" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="xLabel">
            <property name="text">
             <string>X</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDial" name="noiseCenterXDial"/>
          </item>
          <item>
           <widget class="QLabel" name="yLabel">
            <property name="text">
             <string>Y</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDial" name="noiseCenterYDial"/>
          </item>
          <item>
           <widget class="QLabel" name="zLabel">
            <property name="text">
             <string>Z</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDial" name="noiseCenterZDial"/>
          </item>
         </layout>
        </item>
        <item row="9" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="persistenceLabel">
            <property name="text">
             <string>Persistence</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="persistenceSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="persistenceTextbox">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="13" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QLabel" name="colorLabel">
            <property name="text">
             <string>Color</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="colorButton">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="colorRedEdit"/>
          </item>
          <item>
           <widget class="QLineEdit" name="colorGreenEdit"/>
          </item>
          <item>
           <widget class="QLineEdit" name="colorBlueEdit"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="useLightingCheckbox">
       <property name="text">
        <string>Use lighting</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="drawWireframeCheckbox">
       <property name="text">
        <string>Draw wireframe</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="drawNormalsCheckbox">
       <property name="text">
        <string>Draw normals</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="shapesDockEmptySpace" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuToolbars">
    <property name="title">
     <string>&amp;Toolbars</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuToolbars"/>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRevert">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Revert Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionCopy3Dto2D">
   <property name="text">
    <string>Copy &amp;3D tab to 2D tab</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>&amp;Clear Image</string>
   </property>
   <property name="toolTip">
    <string>Clear Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionUseOrbitingCamera">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Use Orbiting Camera</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCopy3Dto2D</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileCopy3Dto2D()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1432</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClear</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>clearImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1555</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRevert</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>revertImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1555</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>checkAllRayFeatures()</slot>
  <slot>uncheckAllRayFeatures()</slot>
  <slot>setAspectRatioText()</slot>
  <slot>fileNew()</slot>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>renderImage()</slot>
  <slot>cancelRender()</slot>
  <slot>fileCopy3Dto2D()</slot>
  <slot>setCameraAxisX()</slot>
  <slot>setCameraAxisY()</slot>
  <slot>setCameraAxisZ()</slot>
  <slot>updateCameraTranslation()</slot>
  <slot>updateCameraRotationN()</slot>
  <slot>updateCameraRotationV()</slot>
  <slot>updateCameraRotationU()</slot>
  <slot>resetUpVector()</slot>
  <slot>updateCameraClip()</slot>
  <slot>updateCameraHeightAngle()</slot>
  <slot>setCameraAxonometric()</slot>
  <slot>clearImage()</slot>
  <slot>revertImage()</slot>
  <slot>resetSliders()</slot>
 </slots>
</ui>
